### PR TITLE
Display .binlog in a tree view

### PR DESCRIPTION
**Pull request by rodrmoya**

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		Message,
+		Diagnostics
+	}
+
+	class BuildOutputNode : TreePosition
+	{
+		public virtual BuildOutputNodeType NodeType { get; set; }
+		public virtual string Message { get; set; }
+		public BuildOutputNode Parent { get; set; }
+		public virtual IList<BuildOutputNode> Children { get; } = new List<BuildOutputNode> ();
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs]: This is still creating a list by default, it shouldn't

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutput.cs [hunk 2]
```diff
@@ -238,4 +249,130 @@ protected override void OnWriteLog (string message)
...
+			this.includeDiagnostics = includeDiagnostics;
+			if (includeDiagnostics) {
+				rootNodes = buildOutput.RootNodes.ToList ();
+			} else {
+				// If not including diagnostics, we need to filter the nodes,
+				// but instead of doing so now for all, we do it on the fly,
+				// as nodes are requested
+				rootNodes = new List<BuildOutputNode> ();
+				foreach (var root in buildOutput.RootNodes) {
+					rootNodes.Add (new FilteredBuildOutputNode (root, null, includeDiagnostics));
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutput.cs]: Shouldn't top level nodes also be filtered?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs (hunk 1 above)
Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs]: Yes, right, was going to do it on a separate PR, but I can add it here

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutput.cs (hunk 2 above)
Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutput.cs]: The nodes to not include when not displaying diagnostics are just messages or empty Task/Target's, so root nodes (Build nodes) are not affected.

**Pull request merged**